examples/vision/mlp_image_classification.py

```diff
@@ -0,0 +1,452 @@
+        self.mlp2 = keras.Sequential(
+            [
+                layers.Dense(units=num_patches),
+                tfa.layers.GELU(),
+                layers.Dense(units=embedding_dim),
+                layers.Dropout(rate=dropout_rate),
+            ]
+        )
+        self.normalize = layers.LayerNormalization(epsilon=1e-6)
+
+    def call(self, inputs):
+        # Apply layer normalization.
+        x = self.normalize(inputs)
+        # Transpose inputs from [num_batches, num_patches, hidden_units] to [num_batches, hidden_units, num_patches].
+        x_channels = tf.linalg.matrix_transpose(x)
+        # Apply mlp1 on each channel independently.
+        mlp1_outputs = self.mlp1(x_channels)
+        # Transpose mlp1_outputs from [num_batches, hidden_dim, num_patches] to [num_batches, num_patches, hidden_units].
+        mlp1_outputs = tf.linalg.matrix_transpose(mlp1_outputs)
+        # Add skip connection.
+        x = mlp1_outputs + inputs
+        # Apply layer normalization.
+        x_patches = self.normalize(x)
+        # Apply mlp2 on each patch independtenly.
+        mlp2_outputs = self.mlp2(x_patches)
+        # Add skip connection.
+        x = x + mlp2_outputs
+        return x
+
+
+"""
+### Build, train, and evaluate an MLP-Mixer model
+
+Note that training the model with the current settings on a V100 GPUs
+takes around 10 seconds per epoch.
+"""
+
+mlpmixer_blocks = keras.Sequential(
+    [MLPMixerLayer(num_patches, embedding_dim, dropout_rate) for _ in range(num_blocks)]
+)
+
+mlpmixer_classifier = build_classifier(mlpmixer_blocks)
+history = run_experiment(mlpmixer_classifier)
+
+"""
+The MLP-Mixer model achieves around 52% accuracy and 81% top-5 accuracy on the test data.
+However, the MLP-Mixer model tends to have much less number of parameters compared
+to convolutional and transformer-based models, which leads to less training and
+serving computational cost.
+
+As mentioned in the [MLP-Mixer](https://arxiv.org/abs/2105.01601) paper,
+when pre-trained on large datasets, or with modern regularization schemes,
+the MLP-Mixer attains competitive scores to state-of-the-art models.
+You can obtain better results by increasing the embedding dimensions,
+increasing, increasing the number of mixer blocks, and training the model for longer.
+You may also try to increase the size of the input images and use different patch sizes.
+"""
+
+"""
+## The FNet model
+
+The FNet uses a similar block to the Transformer block. However, FNet replaces the self-attention layer
+in the Transformer block with a parameter-free 2D Fourier transformation layer:
+
+1. one 1D Fourier Transform is applied along the patches.
+2. one 1D Fourier Transform is applied along the channels.
+"""
+
+"""
+### Implement the FNet module
+"""
+
+
+class FNetLayer(layers.Layer):
+    def __init__(self, num_patches, embedding_dim, dropout_rate, *args, **kwargs):
+        super(FNetLayer, self).__init__(*args, **kwargs)
+
+        self.fourier = layers.Lambda(
```

Comment:
the `Lambda` wrapping here doesn't serve any purpose, I would recommend instead have the fourier transform either as a standalone utility function as a method on the class

Reply:
Done